### ISSUE TITLE
metadata storage

### ISSUE THREAD

**Issue by tphyahoo** [haskoin/haskoin-wallet#129]
this is a stub task the metadata storage work described at #124, and as a precursor to #128.

Currently metadata is going to mysql, and is maybe a little ad hoc.  

Sounds like we might want move to storing pubkey-keyed metadata to flat files, as a precursor to eventual encryption.  For multisig accounts, we key to  DH Merged XPubKey Tuples, as described in #125.

We probably also want to save metadata for labeled transactions, specification wip.

This might be good to do as a BIP, in hopes that it becomes an eventual standard for bip39/bip32 derived wallets to have cross compatible labels and metadata for accounts, transactions, and perhaps other wallet features.

**Comment by tphyahoo** [comment]
Re labeling addresses... see #126 

TLDR, in bip32 wallets, there is a one to one relation address to xpubkey.  So you could just derive the address for every xpubkey, even the ones that don't get used because they're parents.

Con: doesn't support old-style addresses that are not bip32 derived.
Pro: per #128 / #129, we only want to store metadata keyed to xpubkeys anyway.

I suppose non-xpubkey derived derived addresses could also be supported in some ad hoc way, for  non-bip32 compatibility.

**Comment by xenog** [comment]
Yeah, we can always attach metadata to xpubs, even for addresses.  That way we have a uniform key.

**Comment by xenog** [comment]
We can continue this discussion in the new repository if necessary.
